dvc/remote/gdrive.py

```diff
@@ -38,13 +38,34 @@ def __init__(self, path):
         )
 
 
-def gdrive_retry(func):
-    retry_re = re.compile(r"HttpError (403|500|502|503|504)")
+def append_error_codes(exc):
+    import json
+    from pydrive2.files import ApiRequestError
+
+    assert isinstance(exc, ApiRequestError)
+
+    if hasattr(exc, "errors"):
+        return
+
+    exc.errors = None
 
+    if exc.args:
+        http_error = exc.args[0]
+        content = json.loads(http_error.content)
+        if content and "error" in content:
+            exc.errors = content["error"]
+
+
+def gdrive_retry(func):
     def should_retry(exc):
         from pydrive2.files import ApiRequestError
 
-        return isinstance(exc, ApiRequestError) and retry_re.search(str(exc))
+        if not isinstance(exc, ApiRequestError):
+            return False
+
+        append_error_codes(exc)
+        retry_codes = [403, 500, 502, 503, 504]
+        return exc.errors["code"] in retry_codes
```

Comment:
exc.errors can be None, let's make it empty list and use `get("code", None)` ?

Reply:
empty hash

Reply:
Makes sense, thanks!